File: funcscript/Functions/List/SeriesFunction.cs
using FuncScript.Core;
using FuncScript.Model;

namespace FuncScript.Functions.List
{
    public class SeriesFunction : IFsFunction
    {
        public int MaxParsCount => 2;

        public CallType CallType => CallType.Prefix;

        public string Symbol => "Series";

        public int Precedence => 0;

        public object Evaluate(IFsDataProvider parent, IParameterList pars)
        {

            var par0 = pars.GetParameter(parent,0);
            if (!(par0 is int))
                throw new Error.TypeMismatchError($"{this.Symbol}: {ParName(0)} must be an integer");

            int start = (int)par0;
            var par1 = pars.GetParameter(parent,1);

            if (!(par1 is int))
                throw new Error.TypeMismatchError($"{this.Symbol}: {ParName(1)} must be an integer");

            int count = (int)par1;

            var ret = new List<int>();

            for (int i = 0; i < count; i++)
            {
                ret.Add(start + i);
            }

            return new ArrayFsList(ret);
        }

        public string ParName(int index)
        {
            switch(index)
            {
                case 0: return "start";
                case 1: return "count";
                default: return "";
            }
        }
    }
}

File: js-port/walya-js/src/funcs/list/series-function.js
const { BaseFunction, CallType } = require('../../core/function-base');
const { ArrayFsList } = require('../../model/fs-list');
const helpers = require('../helpers');
const { FSDataType } = require('../../core/fstypes');

class SeriesFunction extends BaseFunction {
  constructor() {
    super();
    this.symbol = 'Series';
    this.callType = CallType.Prefix;
  }

  get maxParameters() {
    return 2;
  }

  evaluate(provider, parameters) {
    const error = helpers.expectParamCount(this.symbol, parameters, this.maxParameters);
    if (error) {
      return error;
    }

    const startResult = helpers.requireInteger(this.symbol, parameters.getParameter(provider, 0), 'start');
    if (!startResult.ok) {
      return startResult.error;
    }

    const countResult = helpers.requireInteger(this.symbol, parameters.getParameter(provider, 1), 'count');
    if (!countResult.ok) {
      return countResult.error;
    }

    const start = startResult.value;
    const count = countResult.value;
    const values = [];
    for (let i = 0; i < count; i += 1) {
      values.push(helpers.makeValue(FSDataType.Integer, start + i));
    }
    return helpers.makeValue(FSDataType.List, new ArrayFsList(values));
  }
}

module.exports = {
  SeriesFunction
};
